coalib/bearlib/languages/documentation/DocumentationComment.py

```diff
@@ -32,14 +33,20 @@ def __init__(self, documentation, docstyle_definition,
         :param marker:
             The three-element tuple with marker strings, that identified this
             documentation comment.
-        :param range:
-            The position range of type ``TextRange``.
+        :param position:
+            The starting ``TextPosition`` of the documentation.
         """
         self.documentation = documentation
         self.docstyle_definition = docstyle_definition
         self.indent = indent
         self.marker = marker
-        self.range = range
+        self.position = position
+        self.assemble = None if position is None else self._assemble()
```

Comment:
Don't let the cache grow into nowhere with `maxsize=None`, use a sensible size like `64` or `128`. Though as `assemble()` takes no parameters, `lru_cache` will only cache a single item, thus we can set `maxsize=1` and use `assemble.clear_cache` when modifications occur^^
(Though this is error prone, as we might forget to reset the cache properly^^)